### PR TITLE
[MINOR][SS][DOC] Added missing config `maxFileAge` in file streaming source

### PR DESCRIPTION
**Pull request by linehrr**
## What changes were proposed in this pull request?

added the missing config for structured streaming when using file source.

from the code we have 
```
  /**
   * Maximum age of a file that can be found in this directory, before it is ignored. For the
   * first batch all files will be considered valid. If `latestFirst` is set to `true` and
   * `maxFilesPerTrigger` is set, then this parameter will be ignored, because old files that are
   * valid, and should be processed, may be ignored. Please refer to SPARK-19813 for details.
   *
   * The max age is specified with respect to the timestamp of the latest file, and not the
   * timestamp of the current system. That this means if the last file has timestamp 1000, and the
   * current system time is 2000, and max age is 200, the system will purge files older than
   * 800 (rather than 1800) from the internal state.
   *
   * Default to a week.
   */
  val maxFileAgeMs: Long =
    Utils.timeStringAsMs(parameters.getOrElse("maxFileAge", "7d"))
```
which is not documented. 

also the file processing order was not mentioned but in the code we specifically select the file list based on file mtime: 
```scala
private val fileSortOrder = if (sourceOptions.latestFirst) {
      logWarning(
        """'latestFirst' is true. New files will be processed first, which may affect the watermark
          |value. In addition, 'maxFileAge' will be ignored.""".stripMargin)
      implicitly[Ordering[Long]].reverse
    } else {
      implicitly[Ordering[Long]]
    }

val files = allFiles.sortBy(_.getModificationTime)(fileSortOrder).map { status =>
      (status.getPath.toUri.toString, status.getModificationTime)
    }
```
---------
Screen Shot 2019-05-07 at 5 55 01 PM
---------
Screen Shot 2019-05-07 at 5 54 55 PM

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Hi, @linehrr (ryne.yang). Thank you for your second contribution. Am I right? 
BTW, for the doc change, it's recommended to generate the doc and attach the new screen shot into the PR description.

**Comment by linehrr** [comment]
@dongjoon-hyun 
ok attached pics, edited 2 places regarding file source processing order and `maxFileAge` option.

**Comment by dongjoon-hyun** [comment]
ok to test

**Comment by dongjoon-hyun** [comment]
Thanks, @linehrr .

File: docs/structured-streaming-programming-guide.md [hunk 1]
```diff
@@ -541,6 +541,8 @@ Here are the details of all the sources in Spark.
         <br/>
         <code>fileNameOnly</code>: whether to check new files based on only the filename instead of on the full path (default: false). With this set to `true`, the following files would be considered as the same file, because their filenames, "dataset.txt", are the same:
         <br/>
+        <code>maxFileAge</code>: Maximum age of a file that can be found in this directory, before it is ignored. For the first batch all files will be considered valid. If `latestFirst` is set to `true` and `maxFilesPerTrigger` is set, then this parameter will be ignored, because old files that are valid, and should be processed, may be ignored. The max age is specified with respect to the timestamp of the latest file, and not the timestamp of the current system.(default: a week)
```
Comment (dongjoon-hyun) [inline docs/structured-streaming-programming-guide.md]: According to the attached image, it seems that we need the following changes.
```
- If `latestFirst` is set to `true` and `maxFilesPerTrigger`
+ If <code>latestFirst</code> is set to `true` and <code>maxFilesPerTrigger</code>
```

**Comment by linehrr** [comment]
Fixed the font.

File: docs/structured-streaming-programming-guide.md [hunk 2]
```diff
@@ -510,7 +510,7 @@ returned by `SparkSession.readStream()`. In [R](api/R/read.stream.html), with th
 #### Input Sources
 There are a few built-in sources.
 
-  - **File source** - Reads files written in a directory as a stream of data. Supported file formats are text, csv, json, orc, parquet. See the docs of the DataStreamReader interface for a more up-to-date list, and supported options for each file format. Note that the files must be atomically placed in the given directory, which in most file systems, can be achieved by file move operations.
+  - **File source** - Reads files written in a directory as a stream of data(files will be processed in the order of file's modification time, this is to mimic the time series feature from streaming nature. If `latestFirst` is set, order will be reversed). Supported file formats are text, csv, json, orc, parquet. See the docs of the DataStreamReader interface for a more up-to-date list, and supported options for each file format. Note that the files must be atomically placed in the given directory, which in most file systems, can be achieved by file move operations.
```
Comment (srowen) [inline docs/structured-streaming-programming-guide.md]: I think the text need a number of fixes:

```
Reads files written in a directory as a stream of data. Files will be processed in the order of file modification time. If `latestFirst` is set, order will be reversed. Supported file formats are text, CSV, JSON, ORC, Parquet. See the docs of the DataStreamReader interface for a more up-to-date list, and supported options for each file format. Note that the files must be atomically placed in the given directory, which in most file systems, can be achieved by file move operations.
```

**Review by srowen — CHANGES_REQUESTED** [review]

File: docs/structured-streaming-programming-guide.md (hunk 2 above)
Comment (linehrr) [reply docs/structured-streaming-programming-guide.md]: ok revised sentence.

File: docs/structured-streaming-programming-guide.md [hunk 3]
```diff
@@ -510,8 +510,7 @@ returned by `SparkSession.readStream()`. In [R](api/R/read.stream.html), with th
 #### Input Sources
 There are a few built-in sources.
 
-  - **File source** - Reads files written in a directory as a stream of data. Supported file formats are text, csv, json, orc, parquet. See the docs of the DataStreamReader interface for a more up-to-date list, and supported options for each file format. Note that the files must be atomically placed in the given directory, which in most file systems, can be achieved by file move operations.
-
+  - **File source** - Reads files written in a directory as a stream of data. Files will be processed in the order of file modification time. If `latestFirst` is set, order will be reversed. Supported file formats are text, CSV, JSON, ORC, Parquet. See the docs of the DataStreamReader interface for a more up-to-date list, and supported options for each file format. Note that the files must be atomically placed in the given directory, which in most file systems, can be achieved by file move operations.
```
Comment (felixcheung) [inline docs/structured-streaming-programming-guide.md]: why are ` text, CSV, JSON, ORC, Parquet` capitalized? I thought file format names are lower cased

Comment (srowen) [reply docs/structured-streaming-programming-guide.md]: Maybe identifiers in code like `csv` would be lowercase, but these formats are acronyms and are properly capitalized when named. CSV = Comma Separated Values, etc.

Comment (linehrr) [reply docs/structured-streaming-programming-guide.md]: do we need to change this ?

Comment (srowen) [reply docs/structured-streaming-programming-guide.md]: I don't believe so.

**Review by srowen — APPROVED** [review]

File: docs/structured-streaming-programming-guide.md [hunk 4]
```diff
@@ -541,6 +540,8 @@ Here are the details of all the sources in Spark.
         <br/>
         <code>fileNameOnly</code>: whether to check new files based on only the filename instead of on the full path (default: false). With this set to `true`, the following files would be considered as the same file, because their filenames, "dataset.txt", are the same:
         <br/>
+        <code>maxFileAge</code>: Maximum age of a file that can be found in this directory, before it is ignored. For the first batch all files will be considered valid. If <code>latestFirst</code> is set to `true` and <code>maxFilesPerTrigger</code> is set, then this parameter will be ignored, because old files that are valid, and should be processed, may be ignored. The max age is specified with respect to the timestamp of the latest file, and not the timestamp of the current system.(default: a week)
```
Comment (srowen) [inline docs/structured-streaming-programming-guide.md]: Not a big deal but can you fix the end here, actually: "system. (Default: 1 week)"

Comment (linehrr) [reply docs/structured-streaming-programming-guide.md]: solved.

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to master/branch-2.4.

**Pull request closed without merging**